### PR TITLE
Add TransactionEvent for successful transactions

### LINKED ISSUE #3635: Request: Custom command on verified payment. (/pay) for sounds/titles/anything we want.
### Feature request

**Feature description**
Add a config option to run commands based on verified payment (Once its gone through) for sender and receiver, so we can add sounds for payment and then titles/hotbar titles if we so wish.  

**How the feature is useful**
When chat is moving extremely fast, the payment you have been sent will be lost in seconds, and you have to be eagle hawk eyed to spot it.

Sounds and big titles would help dramatically tell you you just got a payment.

### PR DESCRIPTION
**Pull request by Chew**
This adds an event for plugins to hook into when a transaction occurs, allowing them to do whatever they want.

Closes #3635

Was also considering making it cancellable, may do that if suggested.

bruh

### CODE REVIEW

File: Essentials/src/net/ess3/api/events/TransactionEvent.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+public class TransactionEvent extends Event {
+    private static final HandlerList handlers = new HandlerList();
+
+    private final CommandSource requester;
+    private final IUser target;
+    private final BigDecimal amount;
+
+    public TransactionEvent(CommandSource requester, IUser target, BigDecimal amount) {
+        super();
```
Comment (mdcfe) [inline Essentials/src/net/ess3/api/events/TransactionEvent.java]: Should probably check the current thread here, in case transactions occur async.

Comment (Chew) [reply Essentials/src/net/ess3/api/events/TransactionEvent.java]: Not sure what you mean, doesn't the event get raised after the transaction is over?

File: Essentials/src/com/earth2me/essentials/User.java [hunk 2]
```diff
@@ -11,10 +11,7 @@
 import com.earth2me.essentials.utils.VersionUtil;
 import net.ess3.api.IEssentials;
 import net.ess3.api.MaxMoneyException;
-import net.ess3.api.events.AfkStatusChangeEvent;
-import net.ess3.api.events.JailStatusChangeEvent;
-import net.ess3.api.events.MuteStatusChangeEvent;
-import net.ess3.api.events.UserBalanceUpdateEvent;
+import net.ess3.api.events.*;
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/User.java:14]: Don't use * import please

Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/User.java]: Don't use * imports

**Review by pop4959 — CHANGES_REQUESTED** [review]

File: Essentials/src/com/earth2me/essentials/User.java (hunk 2 above)
Comment (Chew) [reply Essentials/src/com/earth2me/essentials/User.java]: That's just what IntelliJ auto optimize did ;-;

File: Essentials/src/com/earth2me/essentials/User.java [hunk 3]
```diff
@@ -199,6 +196,8 @@ public void payUser(final User reciever, final BigDecimal value, UserBalanceUpda
             reciever.setMoney(reciever.getMoney().add(value), cause);
             sendMessage(tl("moneySentTo", NumberUtil.displayCurrency(value, ess), reciever.getDisplayName()));
             reciever.sendMessage(tl("moneyRecievedFrom", NumberUtil.displayCurrency(value, ess), getDisplayName()));
+            TransactionEvent transactionEvent = new TransactionEvent(this.getSource(), reciever, value);
+            ess.getServer().getPluginManager().callEvent(transactionEvent);
```
Comment (Octol1ttle) [inline Essentials/src/com/earth2me/essentials/User.java]: It's "receiver"

File: Essentials/src/com/earth2me/essentials/User.java (hunk 2 above)
Comment (pop4959) [reply Essentials/src/com/earth2me/essentials/User.java]: you can turn it off

File: Essentials/src/com/earth2me/essentials/User.java (hunk 3 above)
Comment (Chew) [reply Essentials/src/com/earth2me/essentials/User.java]: No it's not

Comment (Chew) [reply Essentials/src/com/earth2me/essentials/User.java]: To clarify: it's spelt that way in the code, not necessarily the proper spelling. I'm not here to fix typos.

Comment (Octol1ttle) [reply Essentials/src/com/earth2me/essentials/User.java]: My inner perfectionist couldn't resist.

Comment (pop4959) [reply Essentials/src/com/earth2me/essentials/User.java]: @l1ttleO Feel free to make a separate PR to fix the typos if you feel the need, but it's not in scope for this one.

File: Essentials/src/com/earth2me/essentials/User.java (hunk 2 above)
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/User.java]:
```suggestion
import net.ess3.api.events.AfkStatusChangeEvent;
import net.ess3.api.events.JailStatusChangeEvent;	
import net.ess3.api.events.MuteStatusChangeEvent;	
import net.ess3.api.events.UserBalanceUpdateEvent;
```

**Review by pop4959 — CHANGES_REQUESTED** [review]

**Review by pop4959 — DISMISSED** [review]

**Review by JRoy — DISMISSED** [review]

**Review by mdcfe — DISMISSED** [review]

**Comment by Chew** [comment]
Oops

**Review by JRoy — DISMISSED** [review]

**Review by JRoy — APPROVED** [review]

**Pull request merged**